### ISSUE TITLE
✨ API - TalentNomination

### ISSUE THREAD

**Issue by petertgiles** [GCTC-NTGC/gc-digital-talent#12293]
## ✨ Feature

Build the API for the talent event nomination models. 

## 🕵️ Details

We need a new model:

1. TalentNomination - represents a submission by a submitter to enter an employee into a talent event

We need several mutations:
1. CreateTalentNomination - new draft model, sets the talent event and submitter association
2. UpdateTalentNomination - update user-changeable fields while draft
3. SubmitTalentNomination - frozen state, must have been completed

## 🎨 Design

@JoshBeveridge 

https://www.figma.com/design/k9lJ77FgLFvpO9L9tVyzID/Talent-nomination-(Employees)?node-id=314-60920&node-type=frame&t=RdnKkm2vLT8XdA2B-0

## 📸 Screenshot

![Image](https://github.com/user-attachments/assets/d9b49d2f-f8b1-4f43-b3a1-2f29c041f8c6)

## 🙋‍♀️ Proposed Implementation

```graphql
type TalentEventNomination {
   id: UUID!
   created_at: Date
   submitted_at: Date
   submitter: User! @belongsTo
   submitter_relationship_to_nominator: Enum
   submitter_relationship_to_nominator_other: String
   nominator: User @belongsTo
   nominator_fallback_work_email: String
   nominator_fallback_name: String
   nominator_fallback_classification: Classification @belongsTo
   nominator_fallback_department: Department @belongsTo
   nominee: User @belongsTo
   nominee_review: Enum
   nominee_relationship_to_nominator: Enum
   nominee_relationship_to_nominator_other: String
   nominate_for_advancement: Boolean
   nominate_for_lateral_movement: Boolean
   nominate_for_dev_programs: Boolean
   advancement_reference: User @belongsTo
   advancement_reference_review: Enum
   advancement_reference_fallback_work_email: String
   advancement_reference_fallback_name: String
   advancement_reference_fallback_classification: Classification @belongsTo
   advancement_reference_fallback_department: Department @belongsTo
   lateral_movement_options: [Enum!]
   lateral_movement_options_other: String
   development_programs: [DevelopmentProgram!] @belongsToMany
   development_program_options_other: String
   nomination_rationale: String
   competencies: [Skill!] @belongsToMany
   nomination_comments: String
   talentNominationEvent: TalentNominationEvent! @belongsTo
  # track progress through the wizard, like pool application
   submitted_steps: [NominationStep!]
}
```

## ✅ Acceptance Criteria

- [x] Only verified employees create a new nomination to a talent event (become submitter)
- [x] Only the submitter can update their draft nomination
- [x] Only the submitter can submit their draft nomination
- [x] No limit on event being open to submit nomination (for now?)
- [x] Only a "complete" draft nomination can be submitted
- [x] Competencies can only be added if the talent event has the option checked (might be classification logic later)
- [x] Only skills from the skill family "Leadership - Executive Behaviours" can be added
- [x] No user can update a submitted nomination (might change in the future)
- [x] Any user can query for all nominations for which they are submitter
- [x] Any user can query for a single nomination by ID for which they are submitter
- [x] No user can query for all nominations
- [ ] Only community admin can query for all submitted nominations for community events in their community (new relationship added to TalentEvent model)
- [x] All model changes (including relationships) tracked in the activity_log table
- [x] local env has seeded rows
- [x] If a submitter has already nominated this person for this event then, this is OK.  No uniqueness constraint.
- [x] If a submitter has already nominated another person for this event then, this is OK.  No uniqueness constraint.
- [x] If a a submitter tries to nominate themselves, this is NOT OK.  Throw an error.
- [x] clicking "start a nomination" on the talent event page creates a draft nomination and navigates to URL /communities/talent-nominations/[nomination ID]/instructions

> [!NOTE]
> Out of scope: attaching to a TalentNominationGroup.  That will happen in #12296 

## 🛑 Blockers

```[tasklist]
### Blocked By
- [ ] #12301
```
